### ISSUE TITLE
Word trigger not being recognized right after arrow key press

### ISSUE THREAD

**Issue by andylima** [federico-terzi/espanso#583]
Hi Federico,

Hopefully I can switch to this text expander!… Congrats for all the amazing work! 🎉

I'm having one major issue…

Most of my triggers have the setting `word: true` — Keeping this in mind: If I press one or more arrow keys (to place the caret somewhere else in a text area/field) and then type a "word" trigger followed by a space, the trigger text should get replaced… but currently doesn't.

Can you fix this?… Thank you so much!

**Comment by federico-terzi** [comment]
Hey @andylima,

Thank you for the feedback! Unfortunately, this is by design and cannot be fixed. Let me explain why this happens:
Espanso can't see the content of the field you are typing into, but only receives a stream of keypresses. If you press an arrow key, as well as a mouse click for example, what happens to the content of the field depends on the application.
Therefore, the only safe choice for espanso is to "reset" the current matching buffer and start from scratch, thus preventing you from completing the trigger.
If you made a mistake while typing the trigger, you should use the "backspace" key to go back and type the letters again, as this behavior is predictable and can be managed by espanso.

Hope this all makes sense!

Cheers :)

**Comment by andylima** [comment]
Sorry, I'm not sure that's the case I'm talking about — I meant, say, pressing the Up and Left arrow keys, and then typing the word trigger (and then Space) and that the word trigger should be expanded. Typinator includes this case. 

By what you wrote, it seems you are talking about using the arrow keys in the middle of typing a trigger?... That's a different case.

**Comment by federico-terzi** [comment]
> I meant, say, pressing the Up and Left arrow keys, and then typing the word trigger (and then Space) and that the word trigger should be expanded. Typinator includes this case.

I see! This makes perfect sense. So you are suggesting using arrow keys as word delimiters right?
I'm currently rewriting the matching core, so this is the perfect time to include such change

Thanks for the explanation :)

**Comment by andylima** [comment]
> (...) So you are suggesting using arrow keys as word delimiters right?

Right. That's it! 🙂

Thanks, Federico!

**Comment by andylima** [comment]
...That said, it's better that espanso keeps expanding — as it currently does — only when typing punctuation characters & line feeds (or the word separators defined in the `word_separators` setting).

What I mean is: espanso shouldn't **expand** on arrow key presses.

...So, we'd assume a key press could be marking the beginning of a word, but (to be on the safe side) not the end (that is, no expansion). If the user really wants expansion in this situation they can just press Space and then delete it.

...Otherwise, whenever we went back to fix some text (say, add one character), we could be triggering expansions inadvertently!

I don't have Typinator installed anymore, but I'm positive that's how it worked for word triggers. **No expansion on arrow key presses.** For now I'm using aText on my Mac — and I can confirm this is the way aText works.

**Comment by federico-terzi** [comment]
> ...So, espanso would assume a key press could be marking the beginning of a word, but (to be on the safe side) not the end (that is, no expansion). If the user really wants expansion in this situation they can just press Space and then delete it.

This seems the best approach, thank you for pointing that out!

**Comment by andylima** [comment]
…I think a mouse left click should also act in this way.

**Comment by federico-terzi** [comment]
@andylima Good point! this is already happening in the current version, although the implementation is not perfect. The next version should greatly improve this aspect :)

**Comment by andylima** [comment]
Thank you so much! 👏
